src/vehicle.cpp

```diff
@@ -1248,6 +1252,152 @@ int vehicle::install_part( int dx, int dy, const vehicle_part &new_part )
     return parts.size() - 1;
 }
 
+bool vehicle::find_rackable_vehicle( std::vector<std::vector<int>> racks_parts )
+{
+    // the racks are already sorted
+    for( auto rack_parts : racks_parts ) {
+        int found_vehicle = -1;
+        std::vector<vehicle *> carry_vehs;
+        carry_vehs.assign( 4, nullptr );
+        vehicle *test_veh = nullptr;
+        std::vector<std::set<tripoint>> test_pts;
+        std::set<tripoint> test_pt;
+        test_pts.assign( 4, test_pt );
+        for( auto rack_part : rack_parts ) {
+            tripoint rack_pos = global_part_pos3( rack_part );
+            for( int i = 0; i < 4; i++ ) {
+                tripoint search_pos( rack_pos + vehicles::cardinal_d[ i ] );
+                test_veh = veh_pointer_or_null( g->m.veh_at( search_pos ) );
+                if( test_veh == this ) {
+                    continue;
+                    // we had a candidate vehicle, this veh is different
+                } else if( carry_vehs[ i ] && test_veh != carry_vehs[ i ] ) {
+                    // is every part of the vehicle adjacent to the rack?
+                    if( test_pts[ i ] == carry_vehs[ i ]->occupied_points ) {
```

Comment:
AFAICT this check isn't reachable unless you are iterating over vehicle A which is adjacent to the rack, then transition to vehicle B which is also adjacent to the rack, i.e.
```
RRR
BAA
```
None of the following scenarios seem to do it:
```
RRR
AAA

RRR
AA

RRR
 AA
```

I think the only sensible place to put this check is immediately after ``test_pts[ i ].insert( search_pos );`` on line 1289.

Reply:
Just as importantly, stepping through this in a debugger and it seems that ``carry_vehs[ i ]->occupied_points`` isn't populated, I'm assuming it's a part of an internal caching scheme and it needs a command to populate these guys before it's accessed.

Reply:
I should be calling get_points() here to force the cache instead of calling it directly. Thanks.

Reply:
Fixed.